muspinsim/cython/parallel.pyx

```diff
@@ -0,0 +1,66 @@
+import numpy as np
+from cython.parallel import prange
+
+cimport numpy as np
+cimport cython
+
+np.import_array()
+
+# Use C versions to avoid linking back to python
+cdef extern from "<math.h>" nogil:
+    double cos(double theta)
+    double sin(double theta)
+
+@cython.boundscheck(False) # Disable bounds-checking
+@cython.wraparound(False)  # Disable negative index wrapping
+def parallel_fast_time_evolve(double [:] times, int other_dimension, double [:, :] A, double [:, :] W):
+    """Computes the result of time evolution of a muon polarisation
+
+    Fast method for computing the result of time evolution of a
+    muon polarisation. To be accurate it requires that
+    \frac{e \hbar^2 B}{2 m_p k T} -> 0 and so is most suitable when T is
+    \inf or B is 0. 
+
+    Calculated as:
+    \frac{1}{2d}\sum_{\alpha, \beta}|<\alpha|\sigma_{\mu}^{\hat{n}}|\beta>|^2
+    +
+    \frac{1}{d}\sum_{\alpha < \beta}|<\alpha|\sigma_{\mu}^{\hat{n}}|\beta>|^2
+        \cos[(E_{\alpha} - E{\alpha})t]
+
+    Arguments:
+        other_dimension {int} -- Value of the dimension labelled as d in the
+                                 equation above. Equal to half the total
+                                 dimension of the system.
+        A {ndarray} -- Matrix giving the amplites of the cosines
+                       |<\alpha|\sigma_{\mu}^{\hat{n}}|\beta>|^2
+        W {ndarray} -- Matrix containing the differences of eigenvalues
+                       computed with np.outer
```

Comment:
A few things that could be changed here:

- `times` missing from `Arguments`
- `amplites` -> `amplitudes`?
- Further clarity of what `A` and `W` are

Possible change below, but happy to go with something else.

```suggestion
    Arguments:
        times {ndarray} -- Times to compute the evolution for, in microseconds
        other_dimension {int} -- Value of the dimension labelled as d in the
                                 equation above. Equal to half the total
                                 dimension of the system.
        A {ndarray} -- Matrix indexed by \alpha and \beta that
                       gives the amplitude
                       |<\alpha|\sigma_{\mu}^{\hat{n}}|\beta>|^2
                       of each cosine term in the sum
        W {ndarray} -- Matrix indexed by \alpha and \beta that contains the
                       differences of their eigenvalues, expressed as angular
                       frequency and computed with np.outer
```